x-pack/plugins/security_solution/public/detections/components/rules/related_integrations/utils.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiLink } from '@elastic/eui';
+import { capitalize } from 'lodash';
+import React from 'react';
+import semver from 'semver';
+import {
+  InstalledIntegration,
+  InstalledIntegrationArray,
+  RelatedIntegration,
+  RelatedIntegrationArray,
+} from '../../../../../common/detection_engine/schemas/common';
+
+/**
+ * Returns an `EuiLink` that will link to a given package/integration/version page within fleet
+ *
+ * @param integration IntegrationDetails describing a package/integrations installed state
+ * @param basePath kbn basepath for composing the fleet URL
+ */
+export const getIntegrationLink = (integration: IntegrationDetails, basePath: string) => {
+  const packageName = integration.package_name;
+  const integrationName = integration.integration_name;
+  const integrationTitle = integration.integration_title ?? integration.package_title;
+  const version = integration.version_satisfied
+    ? integration.package_version
+    : integration.target_version;
+
+  const integrationURL =
+    version !== ''
+      ? `${basePath}/app/integrations/detail/${packageName}-${version}/overview${
+          integrationName ? `?integration=${integrationName}` : ''
+        }`
+      : `${basePath}/app/integrations/detail/${packageName}`;
```

Comment:
I think this calculation should be inside `getInstalledRelatedIntegrations` and we should expose a `target_url` from `IntegrationDetails`. It's a piece of domain logic and it'd be much better to keep it separate from rendering.